### PR TITLE
fixes for vcs_tag

### PR DESCRIPTION
**Pull request by whot**
two doc fixes, one to bail out early if the kwargs don't contain the required bits

### CODE REVIEW

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -2022,6 +2022,8 @@ def func_build_target(self, node, args, kwargs):
             raise InterpreterException('Unknown target_type.')
 
     def func_vcs_tag(self, node, args, kwargs):
+        if 'input' not in kwargs or not 'output' in kwargs:
```
Comment (ignatenkobrain) [inline mesonbuild/interpreter.py]: __[flake8]__
* [E713] test for membership should be 'not in'

File: docs/markdown/Reference-manual.md [hunk 2]
```diff
@@ -635,13 +635,15 @@ Defined tests can be run in a backend-agnostic way by calling `mesontest` inside
...
-This command detects revision control commit information at build time and places it in the specified output file. This file is guaranteed to be up to date on every build. Keywords are similar to `custom_target` and all of them are mandatory.
+This command detects revision control commit information at build time and places it in the specified output file. This file is guaranteed to be up to date on every build. Keywords are similar to `custom_target`.
 
-- `input` file to modify (e.g. `version.c.in`)
-- `output` file to write the results to (e.g. `version.c`)
-- `fallback` version number to use when no revision control information is present, such as when building from a release tarball
+- `input` file to modify (e.g. `version.c.in`) (required)
+- `output` file to write the results to (e.g. `version.c`) (required)
+- `fallback` version number to use when no revision control information is present, such as when building from a release tarball (required)
+- `commmand` string list with the command to execute, see [`custom_target`](#custom_target) for details on how this command must look like
```
Comment (QuLogic) [inline docs/markdown/Reference-manual.md]: Remove 'like'

File: docs/markdown/Reference-manual.md [hunk 3]
```diff
@@ -635,13 +635,15 @@ Defined tests can be run in a backend-agnostic way by calling `mesontest` inside
...
+- `replace_string` string to replace (defaults to `@VCS_TAG@`)
```
Comment (QuLogic) [inline docs/markdown/Reference-manual.md]: I would add "from input file" here to clarify what it's used for.

File: docs/markdown/Reference-manual.md [hunk 4]
```diff
@@ -635,13 +635,15 @@ Defined tests can be run in a backend-agnostic way by calling `mesontest` inside
...
 
-Meson will read the contents of `input`, replace the string `@VCS_TAG@` with the detected revision number and write the result to `output`. This method returns an opaque [`custom_target`](#custom_target) object that you should put in your main program. If you desire more specific behavior than what this command provides, you should use `custom_target`.
+Meson will read the contents of `input`, substitute the replace_string with the detected revision number, and write the result to `output`. This method returns an opaque [`custom_target`](#custom_target) object that can be used as source. If you desire more specific behavior than what this command provides, you should use `custom_target`.
```
Comment (QuLogic) [inline docs/markdown/Reference-manual.md]: Put `replace_string` in backticks.

File: docs/markdown/Reference-manual.md (hunk 2 above)
Comment (whot) [inline docs/markdown/Reference-manual.md]: tried that but it doesn't quite sound like correct english, reworded the sentence instead

File: docs/markdown/Reference-manual.md (hunk 3 above)
Comment (whot) [inline docs/markdown/Reference-manual.md]: reworded too, thanks

File: docs/markdown/Reference-manual.md [hunk 5]
```diff
@@ -635,13 +635,15 @@ Defined tests can be run in a backend-agnostic way by calling `mesontest` inside
...
 
-This command detects revision control commit information at build time and places it in the specified output file. This file is guaranteed to be up to date on every build. Keywords are similar to `custom_target` and all of them are mandatory.
+This command detects revision control commit information at build time and places it in the specified output file. This file is guaranteed to be up to date on every build. Keywords are similar to `custom_target`.
 
-- `input` file to modify (e.g. `version.c.in`)
-- `output` file to write the results to (e.g. `version.c`)
-- `fallback` version number to use when no revision control information is present, such as when building from a release tarball
+- `input` file to modify (e.g. `version.c.in`) (required)
+- `output` file to write the results to (e.g. `version.c`) (required)
+- `fallback` version number to use when no revision control information is present, such as when building from a release tarball (required)
```
Comment (ignatenkobrain) [inline docs/markdown/Reference-manual.md]: shouldn't we just fallback to `version` specified in `project()`?

Comment (whot) [inline docs/markdown/Reference-manual.md]: that'd be a change in behaviour which is outside of the scope of this patch, but happy to do that otherwise.

Comment (jpakkane) [inline docs/markdown/Reference-manual.md]: Let's do this, no point in having separate MRs for something so obvious.

File: docs/markdown/Reference-manual.md [hunk 6]
```diff
@@ -635,13 +635,15 @@ Defined tests can be run in a backend-agnostic way by calling `mesontest` inside
...
+- `commmand` string list with the command to execute, see [`custom_target`](#custom_target) for details on how this command must be specified
```
Comment (jpakkane) [inline docs/markdown/Reference-manual.md]: One "m" too many. :)

**Comment by whot** [comment]
done and rebased on top of master to fix the conflict

**Comment by jpakkane** [comment]
The docs conflict now. Please also add this to release notes.

**Pull request merged**